reference/api/tasks.md

```diff
@@ -262,3 +272,59 @@ Get a single task.
+:::
+
+### Query parameters
+
+A valid `uids`, `statuses`, `types`, `indexUids`, `canceledBy`, or date(`beforeXAt` or `afterXAt`) field is required.
+
+| Query Parameter        | Description                                                                                                                          |
+| :--------------------- | :----------------------------------------------------------------------------------------------------------------------------------- |
+| **`uids`**             | Delete tasks based on `uid`. Separate multiple `uids` with a comma (`,`). Use `uids=*` for all `uids`                                |
+| **`statuses`**         | Delete tasks based on `status`. Separate multiple `statuses` with a comma (`,`). Use `statuses=*` for all `statuses`                 |
+| **`types`**            | Delete tasks based on `type`. Separate multiple `types` with a comma (`,`). Use `types=*` for all `types`                            |
+| **`indexUids`**        | Delete tasks based on `indexUid`. Separate multiple `uids` with a comma (`,`). Use `indexUids=*` for all `indexUids`. Case-sensitive |
+| **`canceledBy`**       | Delete tasks based on the `canceledBy` field                                                                                         |
+| **`beforeEnqueuedAt`** | Delete tasks before a specified `enqueuedAt` date                                                                                    |
+| **`beforeStartedAt`**  | Delete tasks before a specified `startedAt` date                                                                                     |
+| **`beforeFinishedAt`** | Delete tasks before a specified `finishedAt` date                                                                                    |
+| **`afterEnqueuedAt`**  | Delete tasks after a specified `enqueuedAt` date                                                                                     |
+| **`afterStartedAt`**   | Delete tasks after a specified `startedAt` date                                                                                      |
+| **`afterFinishedAt`**  | Delete tasks after a specified `finishedAt` date                                                                                     |
+
+### Example
+
+<CodeSamples id="delete_task_1" />
+
+#### Response: `200 Ok`
+
+```json
+{
+  "taskUid": 3,
+  "indexUid": null,
+  "status": "enqueued",
+  "type": "taskDeletion",
+  "enqueuedAt": "2021-08-12T10:00:00.000000Z"
+}
+```
+
+::: note
+Since `taskDeletion` is a [global task](/learn/advanced/asynchronous_operations.md#global-tasks), the `indexUid` is `null`.
+:::
+
+You can use this `taskUid` to get more details on the [status of the task](#get-one-task).
+
+### Delete all tasks
+
+You can delete all tasks by using the following filter:
+
+<RouteHighlighter method="DELETE" route="/tasks?statuses=failed,canceled,succeeded" />
```

Comment:
```suggestion
<RouteHighlighter method="DELETE" route="/tasks?indexes=*" />
```

Since one line below we're asking the user to have the right on every index, it seems more logical to send this simpler request what do you think?

Reply:
I wasn't sure which one to use. In the end, I decided to go with `statuses=failed,canceled,succeeded"` because this idk emphasizes the fact that you can only delete tasks with these statuses. But if `indexes=*` is simpler, we can go with it. 
Maybe we could add both? 🤔

Reply:
Oh yes I see your point, honestly it's as you want I think both are good :+1: